src/ucp/tag/rndv.c

```diff
@@ -715,15 +709,17 @@ UCS_PROFILE_FUNC_VOID(ucp_rndv_matched, (worker, rreq, rndv_rts_hdr),
             }
         }
         /* put protocol is allowed - register receive buffer memory for rma */
+        ucs_assert(rndv_rts_hdr->size <= rreq->recv.length);
         ucp_request_recv_buffer_reg(rreq, ucp_ep_config(ep)->key.rma_bw_md_map,
-                                    ucs_min(rreq->recv.length, rndv_rts_hdr->size));
+                                    rndv_rts_hdr->size);
     }
 
     /* The sender didn't specify its address in the RTS, or the rndv mode was
      * configured to PUT, or GET rndv mode is unsupported - send an RTR and
      * the sender will send the data with active message or put_zcopy. */
     ucp_rndv_recv_data_init(rreq, rndv_rts_hdr->size);
-    ucp_rndv_req_send_rtr(rndv_req, rreq, rndv_rts_hdr->sreq.reqptr);
+    ucp_rndv_req_send_rtr(rndv_req, rreq, rndv_rts_hdr->sreq.reqptr,
+                          rndv_rts_hdr->size);
```

Comment:
i think it should be ucs_min here as well

Reply:
we can directly use `rndv_rts_hdr->size` here